source/common/network/connection_impl.cc

```diff
@@ -534,11 +534,11 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
                                            connection_stats_->write_current_);
 }
 
-ClientConnectionImpl::ClientConnectionImpl(
-    Event::Dispatcher& dispatcher, Address::InstanceConstSharedPtr address,
-    const Network::Address::InstanceConstSharedPtr source_address)
-    : ConnectionImpl(dispatcher, address->socket(Address::SocketType::Stream), address, nullptr,
-                     source_address, false, false) {}
+ClientConnectionImpl::ClientConnectionImpl(Event::Dispatcher& dispatcher,
+                                           const Address::InstanceConstSharedPtr remote_address,
```

Comment:
Is this really what we want here? Is the idea that the reference count never changes? If so, why don't we just pass in a const reference to the address instance and avoid the involvement of shared_ptr?

Reply:
I think all of these are just "bugs". They should be const&. Same below. I would change them.